partner_statement/tests/test_outstanding_statement.py

```diff
@@ -93,3 +96,104 @@ def test_customer_outstanding_report_no_wizard(self):
         self.assertIn(
             "bucket_labels", report, "There was an error while compiling the report."
         )
+
+    def init_invoice(self, move_type, partner, invoice_date=None, post=True):
+        move_form = Form(
+            self.env["account.move"].with_context(default_move_type=move_type)
+        )
+        move_form.partner_id = partner
+        move_form.invoice_date = invoice_date or fields.Date.from_string("2019-01-01")
+        with move_form.invoice_line_ids.new() as line:
+            line.name = "Test line"
+            line.price_unit = 100
+        move = move_form.save()
+        if post:
+            move.action_post()
+        return move
+
+    def test_exclude_accounts(self):
+        """Accounts can be excluded with a code selector."""
+        # Arrange
+        partners = self.partner1 | self.partner2
+        wizard = self.wiz.with_context(
+            active_ids=partners.ids,
+        ).create({})
+
+        # Edit one invoice
+        # including a new account
+        # that will be the only one not excluded
+        partner_invoice = self.init_invoice("out_invoice", partners[0])
+
+        account = partner_invoice.line_ids.account_id.filtered(
+            lambda a: a.account_type == wizard.account_type
+        )
+        copy_account = account.copy()
+        partner_invoice.line_ids.filtered(
+            lambda l: l.account_id == account
+        ).account_id = copy_account
+
```

Comment:
> @SirPyTech please, can you help me fix the tests?

```suggestion
        partner_invoice.line_ids.flush_recordset()
```
This should fix the test that is failing right now.

Without this the query is probably getting some outdated data of the move lines so the report won't show them.
Maybe the flushing should probably be done on any model involved by the query? Otherwise the DB might have outdated data.

I'm sorry but right now I don't have time to investigate this further, hope this helps!

Reply:
Thanks

Reply:
Thanks to you for porting all this work!